### PR TITLE
Improve sqlexist template

### PR DESCRIPTION
**Pull request by Aagosh10**
Use .First and .FirstID for sqlExist

### CODE REVIEW

File: entc/gen/template/dialect/sql/query.tmpl [hunk 1]
```diff
@@ -249,11 +249,15 @@ func ({{ $receiver }} *{{ $builder }}) sqlCount(ctx context.Context) (int, error
...
-	n, err := {{ $receiver }}.sqlCount(ctx)
-	if err != nil {
+	{{- if $.HasOneFieldID }}
+	_, err := {{ $receiver }}.FirstID(ctx)
+	{{- else }}
+	_, err := {{ $receiver }}.First(ctx)
+	{{- end }}
+	if err != nil && !IsNotFound(err) {
 		return false, fmt.Errorf("{{ $pkg }}: check existence: %w", err)
 	}
```
Comment (masseelch) [inline entc/gen/template/dialect/sql/query.tmpl:259]:
```suggestion
	switch {
	case IsNotFound(err):
		return false, nil
	case err != nil:
		return false, fmt.Errorf("{{ $pkg }}: check existence: %w", err)
	default:
		return true, nil
	}
```

File: entc/gen/template/dialect/sql/query.tmpl [hunk 2]
```diff
@@ -249,11 +249,15 @@ func ({{ $receiver }} *{{ $builder }}) sqlCount(ctx context.Context) (int, error
...
-	return n > 0, nil
+	return !IsNotFound(err), nil
```
Comment (masseelch) [inline entc/gen/template/dialect/sql/query.tmpl:260]:
```suggestion
```

**Review by masseelch — COMMENTED** [review]
Hey, please see my comment. I think a `switch` is more elegant here. @a8m

File: entc/gen/template/dialect/sql/query.tmpl (hunk 2 above)
Comment (masseelch) [reply entc/gen/template/dialect/sql/query.tmpl:260]: see above

File: entc/gen/template/dialect/sql/query.tmpl (hunk 1 above)
Comment (Aagosh10) [reply entc/gen/template/dialect/sql/query.tmpl]: Thanks @masseelch. I have addressed this and regenerated the files.

File: entc/gen/template/dialect/sql/query.tmpl (hunk 2 above)
Comment (Aagosh10) [reply entc/gen/template/dialect/sql/query.tmpl]: Done, thank you

File: entc/integration/cascadelete/ent/comment_query.go [hunk 3]
```diff
@@ -327,6 +328,7 @@ func (cq *CommentQuery) GroupBy(field string, fields ...string) *CommentGroupBy
 //	client.Comment.Query().
 //		Select(comment.FieldText).
 //		Scan(ctx, &v)
+//
```
Comment (masseelch) [inline entc/integration/cascadelete/ent/comment_query.go:331]: Now, that I see this, can you please go generate with go version 1.18? I'd like to have the step for 1.19 done in a separate PR.

File: entc/gen/template/dialect/sql/query.tmpl [hunk 4]
```diff
@@ -249,11 +249,19 @@ func ({{ $receiver }} *{{ $builder }}) sqlCount(ctx context.Context) (int, error
...
 
 func ({{ $receiver }} *{{ $builder }}) sqlExist(ctx context.Context) (bool, error) {
-	n, err := {{ $receiver }}.sqlCount(ctx)
-	if err != nil {
+	{{- if $.HasOneFieldID }}
+	_, err := {{ $receiver }}.FirstID(ctx)
+	{{- else }}
+	_, err := {{ $receiver }}.First(ctx)
+	{{- end }}
+	switch {
```
Comment (a8m) [inline entc/gen/template/dialect/sql/query.tmpl:257]:
```suggestion
	switch _, err := {{ $receiver }}.First{{ if $.HasOneFieldID }}ID{{ end }}(ctx); {
```

Comment (a8m) [reply entc/gen/template/dialect/sql/query.tmpl:257]: Not sure if you see my suggestion correctly because I see this:
image

But I mean this:
image

**Review by a8m — APPROVED** [review]
Thanks for the contribution, @Aagosh10. Looks good. 

Please, fix the minor suggestion.

File: entc/gen/template/dialect/sql/query.tmpl [hunk 5]
```diff
@@ -249,11 +249,14 @@ func ({{ $receiver }} *{{ $builder }}) sqlCount(ctx context.Context) (int, error
 }
 
 func ({{ $receiver }} *{{ $builder }}) sqlExist(ctx context.Context) (bool, error) {
-	n, err := {{ $receiver }}.sqlCount(ctx)
-	if err != nil {
+	switch _, err := {{ $receiver }}.First{{- if $.HasOneFieldID }}ID{{- end }}(ctx);{
```
Comment (a8m) [inline entc/gen/template/dialect/sql/query.tmpl:252]:
```suggestion
	switch _, err := {{ $receiver }}.First{{ if $.HasOneFieldID }}ID{{ end }}(ctx);{
```
There are no spaces to trim.

**Comment by a8m** [comment]
Thanks for the contribution, @Aagosh10.
